packages/extension-api-explorer/src/ExtensionApiExplorer.tsx

```diff
@@ -84,17 +70,53 @@ const configurator = new ExtensionConfigurator()
 export const ExtensionApiExplorer: FC = () => {
   const match = useRouteMatch<{ specKey: string }>(`/:specKey`)
   const extensionContext = useContext<ExtensionContextData>(ExtensionContext)
+  const [specs, setSpecs] = useState<SpecItems>()
 
-  let chosenSdk: any
+  let sdk: Looker31SDK | Looker40SDK
   if (match?.params.specKey === '3.1') {
-    chosenSdk = extensionContext.core31SDK
-  } else if (match?.params.specKey === '4.0') {
-    chosenSdk = extensionContext.core40SDK
+    sdk = extensionContext.core31SDK
+  } else {
+    sdk = extensionContext.core40SDK
+  }
+
+  /** fetch and compile an API specification to an ApiModel
+   *
+   * @param spec to fetch and compile
+   */
+  async function extFetch(spec: SpecItem) {
+    if (!spec.specURL) return undefined
+    const sdk = extensionContext.core40SDK
+    const [version, name] = spec.specURL?.split('/').slice(-2)
+    const content = await sdk.ok(sdk.api_spec(version, name))
+    // TODO figure out why this crazy step is required
+    let json = JSON.parse(content)
+    if (typeof json === 'string') {
+      json = JSON.parse(json)
+    }
+    json = upgradeSpecObject(json)
+    const api = ApiModel.fromJson(json)
+    return api
   }
 
+  useEffect(() => {
+    /** Load Looker /versions information and retrieve all supported specs */
+    async function loadSpecs() {
+      const versions = await sdk.ok(sdk.versions())
```

Comment:
This results in a data getting loaded multiple times
`if (sdk && !specs) {`

Reply:
Good idea